Add function xml_file_to_parameter

MariaF203 opened the pull request.

Commit:
feat: function xml_file_to_parameter

seb5g commented on packages/pymodaq_gui/src/pymodaq_gui/parameter/ioxml.py:
```diff
@@ -564,6 +564,9 @@ def xml_string_to_parameter_dict(xml_string) -> dict:
     return Parameter.create(**xml_string_to_parameter_dict(xml_string))
 
+def xml_file_to_parameter(xml_file) -> Parameter:
+    return Parameter.create(**xml_file_to_parameter_dict(xml_file))
```
could you add a docstring explaingin what it does

seb5g commented on packages/pymodaq_gui/src/pymodaq_gui/parameter/ioxml.py:
```diff
@@ -564,6 +564,9 @@ def xml_string_to_parameter_dict(xml_string) -> dict:
 def xml_string_to_parameter(xml_string) -> Parameter:
     return Parameter.create(**xml_string_to_parameter_dict(xml_string))
```
could you add a docstring explaingin what it does

seb5g commented on packages/pymodaq_gui/src/pymodaq_gui/parameter/ioxml.py:
```diff
@@ -564,6 +564,9 @@ def xml_string_to_parameter_dict(xml_string) -> dict:
+
 
 def XML_string_to_pobject(xml_string) -> Parameter:
     """
```
could you change the odcstring to say its obsolete and the other method should be used?

Commit:
feat: add docstrings for ioxml functions

seb5g committed:
Refactor XML parameter functions for type hints

Updated function signatures to use 'bytes' type for xml_string parameter. Improved docstrings for clarity.
